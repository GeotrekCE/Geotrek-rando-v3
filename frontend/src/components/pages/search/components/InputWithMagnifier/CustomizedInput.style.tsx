import styled, { css } from 'styled-components';
import Input from 'components/Input/Input.style';
import { borderRadius, desktopOnly, getSpacing } from 'stylesheet';

const CustomizedInput = styled(Input)`
  width: ${getSpacing(54)};
  flex-grow: 2;
  ${desktopOnly(css`
    width: ${getSpacing(58)};
  `)}
  height: ${getSpacing(10)};
  ${desktopOnly(css`
    height: ${getSpacing(12)};
  `)}
  border-radius: 0;
  border-top-left-radius: ${borderRadius.squareButton};
  border-bottom-left-radius: ${borderRadius.squareButton};
`;

export default CustomizedInput;
