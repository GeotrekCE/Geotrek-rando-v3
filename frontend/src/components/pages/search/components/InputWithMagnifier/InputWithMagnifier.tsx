import { Search } from 'components/Icons/Search';
import React, { ChangeEvent, FunctionComponent } from 'react';
import { colorPalette } from 'stylesheet';
import CustomizedInput from './CustomizedInput.style';

interface InputWithMagnifierProps {
  value: string | null;
  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
  onButtonClick: () => void;
}

const InputWithMagnifier: FunctionComponent<InputWithMagnifierProps> = ({
  onChange,
  value,
  onButtonClick,
}) => {
  return (
    <div className="flex flex-row w-full desktop:w-auto">
      <CustomizedInput onChange={onChange} value={value !== null ? value : ''} type="text" />
      <div
        className="w-10 h-10 desktop:h-12 desktop:w-12 bg-primary1 rounded-r-md
        flex justify-center items-center
        active:bg-primary1-light
        cursor-pointer
        "
      >
        <div onClick={onButtonClick}>
          <Search size={24} color={colorPalette.primary2} />
        </div>
      </div>
    </div>
  );
};

export default InputWithMagnifier;
