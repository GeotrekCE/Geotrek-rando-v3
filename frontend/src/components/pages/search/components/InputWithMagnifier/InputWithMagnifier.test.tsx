import React from 'react';
import { render } from 'services/testing/reactTestingLibraryWrapper';
import InputWithMagnifier from '.';

describe('InputWithMagnifier', () => {
  it('renders correctly', () => {
    const inputWithMagnifier = render(
      <InputWithMagnifier
        value={null}
        onChange={() => {
          /* does nothing */
        }}
        onSubmit={() => {
          /* does nothing */
        }}
      />,
    );

    expect(inputWithMagnifier).toMatchSnapshot();
  });
});
