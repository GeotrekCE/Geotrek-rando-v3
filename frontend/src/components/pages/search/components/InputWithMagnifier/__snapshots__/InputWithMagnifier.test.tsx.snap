// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithMagnifier renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action="/search"
        class="flex flex-row w-full desktop:w-auto"
      >
        <label
          class="sr-only"
          for=":r0:"
        >
          Recherche
        </label>
        <input
          class="input !w-30 desktop:!w-55 h-10 desktop:h-12 !rounded-r-none !rounded-l-lg"
          id=":r0:"
          name="text"
          placeholder="Rechercher..."
          type="search"
          value=""
        />
        <button
          class="size-10 desktop:size-12 bg-primary1 rounded-r-md flex justify-center items-center active:bg-primary1-light "
          type="submit"
        >
          <svg
            aria-hidden="true"
            class="text-primary2"
            fill="none"
            height="24"
            viewBox="0 0 22 22"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 8.5a7.5 7.5 0 11-15 0 7.5 7.5 0 0115 0z"
              stroke="currentColor"
              stroke-width="2"
            />
            <path
              d="M14.164 14.25l5.586 5.586"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <span
            class="sr-only"
          >
            Recherche
          </span>
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action="/search"
      class="flex flex-row w-full desktop:w-auto"
    >
      <label
        class="sr-only"
        for=":r0:"
      >
        Recherche
      </label>
      <input
        class="input !w-30 desktop:!w-55 h-10 desktop:h-12 !rounded-r-none !rounded-l-lg"
        id=":r0:"
        name="text"
        placeholder="Rechercher..."
        type="search"
        value=""
      />
      <button
        class="size-10 desktop:size-12 bg-primary1 rounded-r-md flex justify-center items-center active:bg-primary1-light "
        type="submit"
      >
        <svg
          aria-hidden="true"
          class="text-primary2"
          fill="none"
          height="24"
          viewBox="0 0 22 22"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 8.5a7.5 7.5 0 11-15 0 7.5 7.5 0 0115 0z"
            stroke="currentColor"
            stroke-width="2"
          />
          <path
            d="M14.164 14.25l5.586 5.586"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <span
          class="sr-only"
        >
          Recherche
        </span>
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
